openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,839 @@
+    //// Create the primary indicator objects ///
+    /////////////////////////////////////////////
 
-    $(function() {
-        $( '#categoryTabs' ).tabs({
-            collapsible: false,
-            selected: -1,
-            active: false,
-            height: 550
-        });
-    });
+    var svThemes;
+    var riskIndicators;
 
-    function findPrimaryIndicators(pdData, pi) {
-        // Find all of the primary indicators
-        if (pi.some(function(currentValue) {
-            return (pdData.type == currentValue);
-        })) {
-            // Create array of primary indicators
-            pdTempPrimaryIndicator.push(pdData.name);
-            pdTempPrimaryIndicatorLevels[pdData.name.toLowerCase()] = (pdData.level);
-            tempWeight[pdData.name.toLowerCase()] = (pdData.weight);
+    // Find all the nodes of type social vulnerability and risk index in the project def
+    for (var k in projectDef.children) {
+        if (projectDef.children[k].type == "Social Vulnerability Index") {
+            svThemes = projectDef.children[k].children;
+        } else if (projectDef.children[k].type == "Risk Index") {
+            riskIndicators = projectDef.children[k].children;
         }
-        (pdData.children || []).forEach(function(currentItem) {
-            findPrimaryIndicators(currentItem, [pi]);
-        });
     }
 
-    function findCategoryIndicators(pdData, ci) {
-        // Find all of the primary indicators
-        if (ci.some(function(currentValue) {
-            return (pdData.type == currentValue);
-        })) {
-            // Create array of primary indicators
-            pdTempCategoryIndicator.push(pdData.name);
-            tempCatWeight[pdData.name.toLowerCase()] = (pdData.weight);
-            // Create a key pair for each category indicator and its respective child level
-            pdTempPILevel[pdData.name.toLowerCase()] = (pdData.children[0].level);
+    // process each Social Vulnerability Index nodes
+    // Get all the primary indicators
+    for (var i = 0; i < svThemes.length; i++) {
+        for (var e = 0 ; e < svThemes[i].children.length; e++ ) {
+            allPrimaryIndicators.push(svThemes[i].children[e].field);
         }
-        (pdData.children || []).forEach(function(currentItem) {
-            findCategoryIndicators(currentItem, [ci]);
-        });
     }
 
-    function checkElementCategory(elementName) {
-        var foo = pdTempPrimaryIndicatorLevels[elementName];
+    /////////////////////////////////////////
+    //// Compute the theme indicators ////
+    /////////////////////////////////////////
 
-        for(var prop in pdTempPILevel) {
-            if(pdTempPILevel.hasOwnProperty(prop)) {
-                if(pdTempPILevel[prop] === foo) {
-                    return prop;
-                }
+    //build the theme indicator object
+    var themeData = [];
+    indicatorString = [];
+
+    function generateObject(indicatorArray) {
```

Comment:
`Object` sounds a bit generic. Please rename it into something more intuitive.
Furthermore, it looks a bit strange that you have no return value, and instead you manipulate a global `themeData` object. Please add a comment to clarify these aspects.